hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/metrics/SummedSharedCounter.java

```diff
@@ -0,0 +1,52 @@
+package pl.allegro.tech.hermes.management.infrastructure.metrics;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.atomic.DistributedAtomicLong;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.counter.ZookeeperCounterException;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.stream.Collectors.toList;
+
+public class SummedSharedCounter {
+
+    private final List<LoadingCache<String, DistributedAtomicLong>> distributedAtomicLongCaches;
+
+    public SummedSharedCounter(List<CuratorFramework> curatorClients, int expireAfter,
+                               int distributedLoaderBackoff, int distributedLoaderRetries) {
+        this.distributedAtomicLongCaches = curatorClients.stream()
+                .map(client -> buildLoadingCache(client, expireAfter, distributedLoaderBackoff, distributedLoaderRetries))
+                .collect(toList());
+    }
+
+    public long getValue(String path) {
+        return distributedAtomicLongCaches.stream()
+                .map(distAtomicLong -> {
+                    try {
```

Comment:
Could you extract this code to private method and name it? In my opinion it would increase a readability :)

Reply:
Done